kv/memberlist/tcp_transport_test.go

```diff
@@ -95,3 +95,15 @@ func TestFinalAdvertiseAddr(t *testing.T) {
 		})
 	}
 }
+
+func TestNonIPsAreRejected(t *testing.T) {
+	cfg := TCPTransportConfig{BindAddrs: flagext.StringSlice{"localhost"}}
+	_, err := NewTCPTransport(cfg, nil, nil)
+	if err == nil {
+		t.Fatal("expected non-nil err, got nil")
+	}
+	want := `could not parse bind addr "localhost" as IP address`
+	if err.Error() != want {
+		t.Fatalf("NewTCPTransport with non-IP address BindAddr: got %v, want %v", err, want)
+	}
```

Comment:
We use the `require` package in other tests to make assertions like these.

Reply:
Done